### ISSUE TITLE
Issue with dotnet-ef tool after moving to Dev Drive

### ISSUE THREAD

**Issue by joakimriedel** [dotnet/sdk#33023]

**Comment by baronfel** [comment]
This is a great report, thank you! I think what's happened is that the tool resolver cache points the various local tool installs to the packages directory, and so when you reconfigured that we didn't update our caches. @dsplaisted we should probably chat about mitigations here - perhaps the resolver cache file should store the observed global packages directory and invalidate of that has changed?

**Comment by dsplaisted** [comment]
Yes, seems like we should be doing something here to detect that the cache is invalid.

**Comment by andrewlangemann** [comment]
I also encountered this issue, but resolved it by deleting the `toolResolverCache` folder per [this answer](https://github.com/dotnet/sdk/issues/11432#issuecomment-933208244) on #11432

**Comment by bricelam** [comment]
I hit this with `docfx`. Deleting the cache resolved it. Diffing the old cache file and the new cache file, it looks like the old cache had some hard-coded references to the old NUGET_PACKAGES directory.

In other words, you should be able to repro this by...

1. Install and run a tool
2. Change NUGET_PACKAGES
3. Try to restore and run the tool

**Comment by baronfel** [comment]
I'm going to close this as a duplicate of https://github.com/dotnet/sdk/issues/11432, but starting in .NET 10 preview 6 you can workaround this kind of failure mode by using the new `dotnet tool exec` one-shot mode to run your tools - it'll acquire them on-demand (unless you already have them locally) and prevent this kind of stale-location mixup. We _should_ still fix the root issue eventually, however.
